docs/sequence/PISPLinkingApi.puml

```diff
@@ -0,0 +1,577 @@
+  scopes: [{
+    accountId: 'dfspa.username.1234',
+    actions: ['accounts.getBalance', 'accounts.transfer'],
+    accountId: 'dfspa.username.5678',
+    actions: ['accounts.getBalance', 'accounts.transfer'],
+  }]
+}
+end note
+Switch --> DFSP_LM: 202 ACCEPTED
+deactivate DFSP_LM
+activate PISP_TP_IN
+Switch ->  PISP_TP_IN: POST /consents
+PISP_TP_IN -> PISP_TP_IN: const model = await loadFromKVS({key: 6789})
+activate PISP_LM
+PISP_TP_IN -> PISP_LM: model.registerCredential()
+PISP_TP_IN --> Switch: 202 Accepted
+deactivate PISP_TP_IN
+deactivate Switch
+rnote right of PISP_LM: state: consentRecieved
+rnote right of PISP_LM #LightBlue
+It's important to save the model with the consentId from this point onwards!
+Requests will not contain a consentRequestId in the upcoming requests.
+This may require changing `PersistentModel` to have a secondary key it can
+store the model with.
+end note
+PISP_LM --> PISP_LM: const challenge = deriveChallenge(consentRequest)
+PISP_LM --> PISP_LM: this.saveToKVS({key: '1a2b3c4d'})
+
+PISP_LM -> PISP_LM: ThirdpartyRequests.putConsents()
+rnote right of PISP_LM: state: signedConsentSent
+activate Switch
+PISP_LM -> Switch: PUT /consents/1a2b3c4d
+
+rnote right of PISP_LM #LightBlue
+PUT /consents/1a2b3c4d
+FSIOP-Source: pispa
+FSIOP-Destination: dfspa
+{
+  scopes: [{
+    accountId: 'dfspa.username.1234',
+    actions: ['accounts.getBalance', 'accounts.transfer'],
+    accountId: 'dfspa.username.5678',
+    actions: ['accounts.getBalance', 'accounts.transfer'],
+  }],
+  credential: {
+    credentialType: "FIDO",
+    status: "PENDING",
+    payload: PublicKeyCredential
+  }
+}
+end note
+
+Switch --> PISP_LM: 202 Accepted
+deactivate PISP_LM
+Switch -> DFSP_TP_IN: PUT /consents/1a2b3c4d
+activate DFSP_TP_IN
+
+DFSP_TP_IN -> DFSP_TP_IN: const model = await loadFromKVS({key: 1a2b3c4d})
+activate DFSP_LM
+rnote right of DFSP_TP_IN: state: signedConsentRecieved
+DFSP_TP_IN -> DFSP_LM: model.validateSignedConsent()
+rnote right of DFSP_TP_IN: state: pendingRegistrationAndValidation
+DFSP_TP_IN --> Switch: 202 Accepted
+deactivate Switch
+deactivate DFSP_TP_IN
+DFSP_LM -> DFSP_LM: Check signed consent.
+rnote right of DFSP_LM #LightBlue
+Does this need a backend request or can the scheme adapter check the consent.
+end note
+rnote right of DFSP_LM: state: signedConsentChecked
+DFSP_LM-> DFSP_LM: model.validateWithAuthService()
+activate Switch
+DFSP_LM -> Switch: POST /consents/1a2b3c4d
+
+rnote left of DFSP_LM #LightBlue
+POST /consents/1a2b3c4d
+FSIOP-Source: dfspa
+FSIOP-Destination: central-auth
+{
+  scopes: [{
+    accountId: 'dfspa.username.1234',
+    actions: ['accounts.getBalance', 'accounts.transfer'],
+    accountId: 'dfspa.username.5678',
+    actions: ['accounts.getBalance', 'accounts.transfer'],
+  }],
+  credential: {
+    credentialType: "FIDO",
+    status: "PENDING",
+    payload: PublicKeyCredential
+  }
+}
+end note
+Switch --> DFSP_LM: 202 Accepted
+deactivate DFSP_LM
```

Comment:
drop `deactivate DFSP_LM` here to express the model is waiting on `PUT /consents/1a2b3c4d` from auth service OR use `saveToKVS()`to express preserving the state